+bisos/+iteration/+step/convergence.m

```diff
@@ -0,0 +1,107 @@
+classdef convergence < bisos.iteration.Step
+   
+properties
+    type = 'convergence';
+    varout = [];
+    varin;
+    
+    levels;
+    
+    % default properties for convergence verification
+    options = struct('ctol', 10^-9, 'domain', [-1 1]);
+    
+end
+
+methods
+    function step = convergence(prob, vars, varargin)
+        % new level set convergence setup
+        
+        step.varin = {};
+        
+        % get name of variables that the step needs to start
+        for v = vars(:)'
+            if isa(v{:}, 'char')
+                assert(hasvariable(prob,v), 'Unknown variable ''%s''.', v{:});
+                step.varin(end+1) = v;
+            end
+        end
+        
+        % reshape level sets into cell matrix L
+        % where L{i,1} is the i-th polynomial and L{i,2} is the i-th level
+        step.levels = reshape(vars,[],2)';
+        
+        % assign options
+        varargin = [varargin{:}];
```

Comment:
perhaps it would be better to just accept options as name-value pairs, not nested in an additional cell array.

Reply:
Not sure what you mean by that, are you saying to not have a single variable that stores `varargin{1}` and another for `varargin{2}`, but only a single variable that has both